### PR TITLE
BCDA-3879 - Introduce CLI command to remove data from failed jobs

### PR DESCRIPTION
**Pull request by mtrang1263**
### Fixes [BCDA-3879](https://jira.cms.gov/browse/BCDA-3879)

During our audit of the [job cleanup process](https://jira.cms.gov/browse/BCDA-1651), we discovered that failed job data is kept around. We need to add a process to remove the data associated with failed jobs.

### Change Details
1. Add new cli command `cleanup-failed` that will remove data in the PAYLOAD and STAGING dirs.
2. Introduce new job status, `FailedExpired` that will indicate that the data associated with the failed job is returned.
3. Update the argument for threshold to an integer (from string). Previously, we were immediately parsing the string to an integer.
### Security Implications

- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [x] new data stored or transmitted
When executed, it will cleanup any data associated with a failed job.
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change

### Acceptance Validation
1. Added unit test to verify that we can cleanup data associated with failed jobs.
2. Ran a [Jenkins job](https://bcda-ci.adhocteam.us/job/BCDA%20-%20Archive%20and%20Clean%20Job%20Data/891/) to confirm that the process works on the dev environment.

### CODE REVIEW

**Comment by mtrang1263** [comment]
The commit history is all messed up since I started this work on the GORM removal branch. The files changed reflect what was updated in this PR. 

Since we do squash merges, this messy history will be cleaned up when we merge this PR.

File: bcda/bcdacli/cli.go [hunk 1]
```diff
@@ -331,39 +331,51 @@ func setUpApp() *cli.App {
...
+					Destination: &thresholdHr,
 				},
 			},
 			Action: func(c *cli.Context) error {
-				th, err := strconv.Atoi(threshold)
-				if err != nil {
-					return err
-				}
-				return archiveExpiring(th)
+				cutoff := time.Now().Add(-time.Hour * time.Duration(thresholdHr))
```
Comment (adamayd) [inline bcda/bcdacli/cli.go:343]: nit - match syntax on line 359 & 376

**Review by adamayd — DISMISSED** [review]
LGTM!

**Review by adamayd — APPROVED** [review]

**Comment by adamayd** [comment]
LGTM

**Pull request merged**